### PR TITLE
Migrate React Query to v4

### PR DESCRIPTION
**Pull request by Belco90**
## Changes

- Migrate React Query to v4

## Context

Since v4 belongs to the `@tankstack/` scope, Renovate wasn't aware of the new version. I've migrated it manually.

Some advantages of v4:

- [ Optimized for modern browsers](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#supported-browsers) (which also means smaller package size)
- [Support for React v18](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#support-for-react-18)
- [Offline support](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#proper-offline-support)

## Tests

I've checked my work by:

- [ ] Adding new tests or adjusting existing tests
- [X] Testing manually

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "type-check": "tsc --project tsconfig.validation.json",
     "type-check:cypress": "tsc --project cypress/tsconfig.json",
     "lint": "eslint . --ext .js,.jsx,.ts,.tsx --max-warnings 0 --report-unused-disable-directives",
-    "lint:fix": "pnpm lint -- --fix",
```
Comment (Belco90) [inline package.json:18]: The standalone `--` was necessary for npm, but it's not for pnpm.

**Comment by HonkingGoose** [comment]
> Since v4 belongs to the `@tankstack/` scope, Renovate wasn't aware of the new version. I've migrated it manually.

Thanks for telling me. I'm fixing this problem upstream at the `renovatebot/renovate` repository. Can you check my issue, and check if I'm getting the correct upgrade path?

- https://github.com/renovatebot/renovate/issues/17161

File: package.json (hunk 1 above)
Comment (HonkingGoose) [reply package.json:18]: Cool! We can drop the confusing double `--`! 🥳

**Review by HonkingGoose — APPROVED** [review]
Our "repository search bar" on the Vercel preview still works. I guess that means this PR is good to go?

**Comment by Belco90** [comment]
> Our "repository search bar" on the Vercel preview still works. I guess that means this PR is good to go?

Yeah, basically! Happo caught a problem where the spinner was always displayed, but I fixed it by using `isFetching` instead of `isLoading`. Everything else keeps working the same way.

**Pull request merged**